httpserver.lua

```diff
@@ -139,16 +147,25 @@ return function (port)
                if connectionThreadStatus == "suspended" then
                   -- Not finished sending file, resume.
                   local status, err = coroutine.resume(connectionThread)
-                  if not status then log(connection:getpeer(), "Error: " .. err) end
+                  if not status then
+                     log(connection, "Error: "..err)
+                     log(connection, "closing connetion", "error")
+                     connection:close()
+                     connectionThread = nil
+                     collectgarbage()
+                  end
                elseif connectionThreadStatus == "dead" then
                   -- We're done sending file.
+                  log(connection, "closing connetion","thread is dead")
                   connection:close()
                   connectionThread = nil
+                  collectgarbage()
                end
             end
          end
 
          local function onDisconnect(connection, payload)
+            print("disconnected")

```

Comment:
hmm this is only printed if the client disconnects, so to get the bracket open / close I like it.
But I can still take it out